master/pkg/searcher/simulate.go

```diff
@@ -1,222 +1,121 @@
-			if err != nil {
-				return simulation, err
-			}
-			nextTrialID++
-		case ValidateAfter:
-			simulation.Results[requestID] = append(simulation.Results[requestID], operation)
-			s.SetTrialProgress(requestID, PartialUnits(operation.Length))
-
-			metric := valFunc(random, trialIDs[requestID], trialOpIdxs[requestID])
-			ops, err := s.ValidationCompleted(requestID, metric, operation)
-			if err != nil {
-				return simulation, err
-			}
-			trialOpIdxs[requestID]++
-
-			shutdown, err = handleOperations(pending, &requestIDs, ops)
-			if err != nil {
-				return simulation, err
-			}
-		case Close:
-			delete(pending, requestID)
-			ops, err := s.TrialClosed(requestID)
-			if err != nil {
-				return simulation, err
-			}
-			shutdown, err = handleOperations(pending, &requestIDs, ops)
-			if err != nil {
-				return simulation, err
-			}
-		default:
-			return simulation, errors.Errorf("unexpected searcher operation: %T", operation)
-		}
-		if shutdown {
-			if len(pending) != 0 {
-				return simulation, errors.New("searcher shutdown prematurely")
-			}
-			break
-		}
-
-		progress := s.Progress()
-		if progress < lastProgress {
-			return simulation, errors.Errorf(
-				"searcher progress dropped from %f%% to %f%%", lastProgress*100, progress*100)
-		}
-		lastProgress = progress
+// Proto converts the TrialSummary to its protobuf representation.
+func (rs TrialSummary) Proto() *experimentv1.TrialSummary {
+	return &experimentv1.TrialSummary{
+		Count: int32(rs.Count),
+		Unit:  rs.Unit.Proto(),
 	}
+}
 
-	lastProgress = s.Progress()
-	if lastProgress != 1.0 {
-		return simulation, errors.Errorf(
-			"searcher progress was not equal to 100%%: %f%%", lastProgress*100)
+// Proto converts the SearchSummary to its protobuf representation.
+func (s SearchSummary) Proto() *experimentv1.SearchSummary {
+	var trialSummaries []*experimentv1.TrialSummary
+	for _, v := range s.Trials {
+		trialSummaries = append(trialSummaries, v.Proto())
 	}
-	if len(simulation.Results) != len(requestIDs) {
-		return simulation, errors.New("more trials created than completed")
+	return &experimentv1.SearchSummary{
+		Config: protoutils.ToStruct(s.Config),
+		Trials: trialSummaries,
 	}
-	return simulation, nil
 }
 
-func handleOperations(
-	pending map[model.RequestID][]Operation, requestIDs *[]model.RequestID, operations []Operation,
-) (bool, error) {
-	for _, operation := range operations {
-		switch op := operation.(type) {
-		case Create:
-			*requestIDs = append(*requestIDs, op.RequestID)
-			pending[op.RequestID] = []Operation{op}
-		case Requested:
-			pending[op.GetRequestID()] = append(pending[op.GetRequestID()], op)
-		case Shutdown:
-			return true, nil
-		default:
-			return false, errors.Errorf("unexpected operation: %T", operation)
-		}
+// Simulate generates the intended training plan for the searcher.
+func Simulate(conf expconf.SearcherConfig, hparams expconf.Hyperparameters) (SearchSummary, error) {
```

Comment:
well, ASHA is no longer controlled by the searcher sending `ValidateAfter` workloads, so there was no easy way to implement this same behavior. like, it's up to users to validate at specified intervals and the searcher will try and roughly stop on rung boundaries. 

i guess this simulate is less rigorous because the implementation is less rigorous.